phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/apps/ApplicationFilter.java

```diff
@@ -117,6 +138,53 @@ public void clearAll() {
     }
 
     /**
+     * Filter function for {@link CategoryDTO} objects
+     *
+     * @param category The category which should be checked
+     * @return True if the given <code>category</code> fulfills the filter condition, false otherwise
+     */
+    public boolean filter(CategoryDTO category) {
+        boolean result = true;
+
+        /*
+         * If "commercial" is not selected don't show commercial games
+         */
+        if (!containCommercialApplications.getValue()) {
+            result &= category.getApplications().stream()
+                    .anyMatch(application -> application.getScripts().stream().anyMatch(script -> script.isFree()));
+        }
+
+        /*
+         * If "Requires patch" is selected show show games that require a patch to run (e.g. no CD)
+         */
+        if (containRequiresPatchApplications.getValue()) {
+            result &= category.getApplications().stream().anyMatch(
+                    application -> application.getScripts().stream().anyMatch(script -> !script.isRequiresPatch()));
+        }
+
+        /*
+         * If "Testing" is not selected don't show games that are currently in a testing stage
```

Comment:
missing comma: If "Testing" is not selected, don't show games that are currently in a testing stage